tools/rtpsrelay/GuidAddrSet.cpp

```diff
@@ -66,31 +66,24 @@ GuidAddrSet::record_activity(const Proxy& proxy,
   addr_set_stats.deactivation = deactivation;
   relay_participant_status_reporter_.set_alive_active(proxy, src_guid, true, true);
 
-  {
-    AddrSet& addr_set = *addr_set_stats.select_addr_set(remote_address.port, now);
-    const auto pos = addr_set.find(remote_address);
-    if (pos != addr_set.end()) {
-      pos->second = expiration;
-    } else if (config_.max_addr_set_size() == 0 || addr_set.size() < config_.max_addr_set_size()) {
-      addr_set[remote_address] = expiration;
-      if (config_.log_activity()) {
-        ACE_DEBUG((LM_INFO, "(%P|%t) INFO: GuidAddrSet::record_activity "
-                   "%C %C is at %C %C into session addrs=%B total=%B remote=%B deactivation=%B expire=%B admit=%B\n",
-                   handler.name().c_str(),
-                   guid_to_string(src_guid).c_str(),
-                   OpenDDS::DCPS::LogAddr(remote_address.addr).c_str(),
-                   addr_set_stats.get_session_time(now).sec_str().c_str(),
-                   addr_set.size(),
-                   guid_addr_set_map_.size(),
-                   remote_map_.size(),
-                   deactivation_guid_queue_.size(),
-                   expiration_guid_addr_queue_.size(),
-                   admission_control_queue_.size()));
-      }
-      relay_stats_reporter_.new_address(now);
-      const GuidAddr ga(src_guid, remote_address);
-      expiration_guid_addr_queue_.push_back(std::make_pair(expiration, ga));
+  if (addr_set_stats.upsert_address(remote_address, now, expiration, config_.max_addr_set_size())) {
```

Comment:
Does the config's `max_addr_set_size` also need to be renamed?

Reply:
Probably